batch.go

```diff
@@ -0,0 +1,126 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+
+	"filippo.io/edwards25519"
+)
+
+// Verifier holds entries of public keys, signature and a scalar which are used for batch verification.
+type Verifier struct {
```

Comment:
In the `ed25519-zebra` implementation, all of the batch verification API is in the `batch` submodule, so users can refer to it as `batch::Verifier`.  But since Go's naming system is not like that, and from running `godoc`, everything will get rendered together:
<img width="575" alt="Screen Shot 2021-01-25 at 4 03 37 PM" src="https://user-images.githubusercontent.com/44879/105781603-b16ce880-5f27-11eb-87ad-6a552eef6516.png">

So instead of calling it `Verifier`, I think it would be more clear if it were `BatchVerifier`:
```suggestion
type BatchVerifier struct {
```

Then there's no confusion about "how do I verify a single signature with a `Verifier`?" etc, because it's clearly a batch-only API.